test/core.js

```diff
@@ -67,6 +67,48 @@ describe('snabbdom', function() {
       var vnode = h('a', {}, 'I am a string');
       assert.equal(vnode.text, 'I am a string');
     });
+    it('can create vnode with null props', function() {
+      var vnode = h('a', null);
+      assert.deepStrictEqual(vnode.data, {});
+      vnode = h('a', null, ['I am a string']);
+      assert.equal(vnode.children[0].text, 'I am a string');
+    });
+    it('can create vnode with children in vargs ', function() {
+      var vnode = h('a', null,
+        'some content',
+        h('span'),
+        ['part 1', 'part 2'],
+      );
```

Comment:
This is how JSX passes children to the factory, right?

Reply:
that's right, the children will be passed as variable number of ending args.

so 

```
<div>
  <span>Hello</span>
  <span>World</span>
</div>
```

is transpiled as 

```
h("div", null,
    h("span", null, "Hello"),
    h("span", null, "World"));
```

e.g: https://typescript-play.js.org/?jsx=2#code/PQKhAIAECsGcA9wAtwmAKADwBMCWA3APnXHE1gAcBDAO0IAkBTAG2YHtNhLbjTzq6AdTYAnZtk7c6WYHiJA

And that's why we need the flattening because user can do

```ts
<div>
  <span>Hello</span>
  <span>World</span>
  {["one", "two", "three"].map(str => <span>{str}</span>)}
</div>
```

which transpiles to

```
h("div", null,
    h("span", null, "Hello"),
    h("span", null, "World"),
    ["one", "two", "three"].map(str => h("span", null, str)));
```